ext/ffi/lib.rs

```diff
@@ -996,19 +996,24 @@ fn ffi_parse_buffer_arg(
   // 5. Null: Very uncommon / can be represented by a 0.
 
   let pointer = if let Ok(value) = v8::Local::<v8::ArrayBuffer>::try_from(arg) {
-    let backing_store = value.get_backing_store();
-    &backing_store[..] as *const _ as *const u8
+    value.data()
   } else if let Ok(value) = v8::Local::<v8::ArrayBufferView>::try_from(arg) {
     let byte_offset = value.byte_offset();
-    let backing_store = value
+    let pointer = value
       .buffer(scope)
       .ok_or_else(|| {
         type_error("Invalid FFI ArrayBufferView, expected data in the buffer")
       })?
-      .get_backing_store();
```

Comment:
This optimisation of picking out the data pointer directly from the `ArrayBuffer` instead of getting the `BackingStore` and taking the data pointer from there is a double-edged sword for externally backed zero-sized buffers:
1. This method will return a null pointer for those, even though the backing store pointer is something else. So, essentially this breaks getting the pointer out of externally backed zero-sized buffers.
2. Fast API calls likewise get a null pointer for externally backed zero-sized buffers. So, this essentially unifies how our FFI calls work with externally backed zero-sized buffers, removing a potentially very confusing bug.